tests/selenium/conftest.py

```diff
@@ -0,0 +1,196 @@
+        self.mixin = mixin
+
+    @preconditions(_validate_row, _validate_col_id, _validate_state)
+    def get(self, row, col_id, state = _READY):
+        self.mixin._wait_for_table(self.id, state)
+
+        return self.mixin.find_elements(
+            '#{} {} tbody tr th.dash-header[data-dash-column="{}"]'.format(self.id, state, col_id)
+        )[row]
+
+    @preconditions(_validate_row, _validate_col_id, _validate_state)
+    def hide(self, row, col_id, state = _READY):
+        self.get(row, col_id, state).find_element_by_css_selector(
+            '.column-header--hide'
+        ).click()
+
+    @preconditions(_validate_row, _validate_col_id, _validate_state)
+    def sort(self, row, col_id, state = _READY):
+        self.get(row, col_id, state).find_element_by_css_selector(
+            '.column-header--sort'
+        ).click()
+
+
+class DataTableFacade(object):
+    @preconditions(_validate_id, _validate_mixin)
+    def __init__(self, id, mixin):
+        self.id = id
+        self.mixin = mixin
+
+        self.cell = DataTableCellFacade(id, mixin)
+        self.column = DataTableColumnFacade(id, mixin)
+
+    def click_next_page(self):
+        self.mixin._wait_for_table(self.id)
+
+        self.mixin.find_element(
+            '#{} button.next-page'.format(self.id)
+        ).click()
+
+    def click_prev_page(self):
+        self.mixin._wait_for_table(self.id)
+
+        self.mixin.find_element(
+            '#{} button.previous-page'
+        ).click()
+
+
+class DataTableMixin(object):
+    @preconditions(_validate_id, _validate_state)
+    def _wait_for_table(self, id, state = _ANY):
+        return WebDriverWait(self.driver, _TIMEOUT).until(
+            EC.presence_of_element_located((By.CSS_SELECTOR, '#{} {}'.format(id, state)))
+        )
+
+    @preconditions(_validate_id)
+    def table(self, id):
+        return DataTableContext(
+            DataTableFacade(id, self)
+        )
+
+    def copy(self):
+        ActionChains(self.driver).key_down(Keys.CONTROL).send_keys('c').key_up(
+            Keys.CONTROL
+        ).perform()
+
+    def paste(self):
+        ActionChains(self.driver).key_down(Keys.CONTROL).send_keys('v').key_up(
+            Keys.CONTROL
+        ).perform()
+
+
+    @preconditions(_validate_key)
+    def hold(self, key):
+        return HoldKeyContext(self, key)
+
+
+class DataTableComposite(Browser, DataTableMixin):
+    def __init__(self, server, **kwargs):
+        super(DataTableComposite, self).__init__(**kwargs)
+        self.server = server
+
+        self.READY = _READY
+        self.LOADING = _LOADING
+        self.ANY = _ANY
+
+    def start_server(self, app, **kwargs):
+        """start the local server with app"""
+
+        # start server with app and pass Dash arguments
+        self.server(app, **kwargs)
+
+        # set the default server_url, it implicitly call wait_for_page
+        self.server_url = self.server.url
+
+
+@pytest.fixture
+def test(request, dash_thread_server, tmpdir):
```

Comment:
This is the `dash_duo`, `dash_dcc` equivalent for the table. As testing the table is mostly about interacting with discreet items inside the table component itself, this mixin aims to expose in an easy to use way the various parts of the table. To begin: cell and column headers.

It puts into application some lessons learned from the Cypress/JS helper so as to minimize finicking. For example, when one passes an integer vs a string for a cell column name the helper will figure out what the right selector is. Also, a lot of validations have to do with getting the text content of the cells, this is also streamlined so that it's no longer necessary to click the cell / know that the cell is editable or readonly, etc.